src/os/bluestore/BlueStore.cc

```diff
@@ -8550,8 +8550,10 @@ void BlueStore::_kv_sync_thread()
 	} else if (deferred_aggressive) {
 	  force_flush = true;
 	}
-      } else
-	force_flush = true;
+      } else {
+      	if(aios > 0 || deferred_done.size() > 0)
```

Comment:
I'd prefer `if (aios || !deferred_done.empty())`

Reply:
Actually this is not right; deferred_done.empty() does not imply that there is no inflight aios

Reply:
aios > 0 is already taken into consideration on 1st if statement

Reply:
Also All if statements are doing `force_flush = true`
We can also combine them using `||`. This way all checks would be done without falling into different statements.
Something as:
      if (bluefs_single_shared_device && bluefs) {
        if (aios || 
	   (kv_committing.empty() && kv_submitting.empty() && deferred_stable.empty()) ||
	   !deferred_done.empty()) {
             force_flush = true;
        }
      } else
        force_flush = true;